drake/solvers/NonlinearProgramSnoptmex.cpp

```diff
@@ -16,6 +17,14 @@ using namespace std;
 
 static char* snopt_userfun_name;
 
+// NOTE: all snopt calls will use this shared memory... so this code is NOT THREAD SAFE
+static unique_ptr<snopt::doublereal []> rw;
+static unique_ptr<snopt::integer []> iw;
+static unique_ptr<char []> cw;
+static snopt::integer lenrw=0;
+static snopt::integer leniw=0;
+static snopt::integer lencw=0;
+
 static int snopt_userfun(snopt::integer *Status, snopt::integer *n, snopt::doublereal x[],
```

Comment:
We should always avoid manual memory management.  That normally means smart pointers.

On Sep 14, 2015, at 2:43 PM, Hongkai Dai <notifications@github.com> wrote:

> In drake/solvers/NonlinearProgramSnoptmex.cpp:
> 
> > @@ -16,6 +17,14 @@ using namespace std;
> >  
> >  static char* snopt_userfun_name;
> >  
> > +// NOTE: all snopt calls will use this shared memory... so this code is NOT THREAD SAFE
> > +static unique_ptr<snopt::doublereal []> rw;
> > +static unique_ptr<snopt::integer []> iw;
> > +static unique_ptr<char []> cw;
> > +static snopt::integer lenrw=0;
> > +static snopt::integer leniw=0;
> > +static snopt::integer lencw=0;
> > +
> >  static int snopt_userfun(snopt::integer *Status, snopt::integer *n, snopt::doublereal x[],
> +1 for making them static.
> 
> What is our preference in using pointer? Should we always prefer smart pointer to raw pointer?
> 
> —
> Reply to this email directly or view it on GitHub.
>